apps/citizen-func/utils/redis_storage.ts

```diff
@@ -4,67 +4,86 @@ import * as TE from "fp-ts/lib/TaskEither";
 import * as redis from "redis";
 
 export const setWithExpirationTask = (
-  redisClient: redis.RedisClientType,
+  redisClientTask: TE.TaskEither<Error, redis.RedisClientType>,
   key: string,
   value: string,
   expirationInSeconds: number,
   errorMsg?: string
-): TE.TaskEither<Error, true> =>
+): TE.TaskEither<Error, boolean> =>
   pipe(
-    TE.tryCatch(
-      () => redisClient.setEx(key, expirationInSeconds, value),
-      () =>
-        new Error(errorMsg ? errorMsg : "Error setting key value pair on redis")
+    redisClientTask,
+    TE.chain(client =>
+      TE.tryCatch(
+        () => client.setEx(key, expirationInSeconds, value),
+        () =>
+          new Error(
+            errorMsg ? errorMsg : "Error setting key value pair on redis"
+          )
+      )
     ),
-    TE.map(_ => true)
+    TE.map(constTrue)
   );
 
 export const setTask = (
-  redisClient: redis.RedisClientType,
+  redisClientTask: TE.TaskEither<Error, redis.RedisClientType>,
   key: string,
   value: string,
   errorMsg?: string
-): TE.TaskEither<Error, true> =>
+): TE.TaskEither<Error, boolean> =>
   pipe(
-    TE.tryCatch(
-      () => redisClient.set(key, value),
-      () =>
-        new Error(errorMsg ? errorMsg : "Error setting key value pair on redis")
+    redisClientTask,
+    TE.chain(client =>
+      TE.tryCatch(
+        () => client.set(key, value),
+        () =>
+          new Error(
+            errorMsg ? errorMsg : "Error setting key value pair on redis"
+          )
+      )
     ),
-    TE.map(_ => true)
+    TE.map(constTrue)
   );
 
 export const deleteTask = (
-  redisClient: redis.RedisClientType,
+  redisClientTask: TE.TaskEither<Error, redis.RedisClientType>,
   key: string
 ): TE.TaskEither<Error, boolean> =>
   pipe(
-    TE.tryCatch(
-      () => redisClient.del(key),
-      () => new Error("Error deleting key value pair on redis")
+    redisClientTask,
+    TE.chain(client =>
+      TE.tryCatch(
+        () => client.del(key),
+        () => new Error("Error deleting key value pair on redis")
+      )
     ),
     TE.map(constTrue)
   );
 
 export const getTask = (
-  redisClient: redis.RedisClientType,
+  redisClientTask: TE.TaskEither<Error, redis.RedisClientType>,
   key: string
 ): TE.TaskEither<Error, O.Option<string>> =>
   pipe(
-    TE.tryCatch(
-      () => redisClient.get(key),
-      () => new Error("Error while retrieving value from redis")
+    redisClientTask,
+    TE.chain(client =>
+      TE.tryCatch(
+        () => client.get(key),
+        () => new Error("Error while retrieving value from redis")
+      )
     ),
     TE.map(O.fromNullable)
```

Comment:
ditto

Reply:
What kind of additional check do we expect here? I also checked the example you provided but the code seems very similar to me https://github.com/pagopa/io-functions-cgn/blob/4df37c956d23a4aa2364a3017e4398bc71a81f18/utils/redis_storage.ts#L84

Reply:
No check, in effect. My fault.